### PR TITLE
Fix layer backed nodes not update properly

### PR DESCRIPTION
**Pull request by wsdwsd0829**

### CODE REVIEW

**Comment by wsdwsd0829** [comment]
Layers can be thrashed by UIKit with (exit then enter) during push/pop view controller or reload cells. 
If exit & cancelAsyncDisplay, we are not showing the content if it's not nil. 
More details are explained here: https://github.com/TextureGroup/Texture/pull/1086

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/35891308e3a54d3f3ffb618bdd501708/raw/fedfc46dba5428194bc186d146f6eecfab643554/log.txt)

File: Source/ASDisplayNode.mm [hunk 1]
```diff
@@ -2845,9 +2845,7 @@ - (void)__exitHierarchy
     _flags.isExitingHierarchy = YES;
     _flags.isInHierarchy = NO;
 
-    [self._locked_asyncLayer cancelAsyncDisplay];
-
-    // Don't call -didExitHierarchy while holding __instanceLock__.
+    // Don't call -didExitHierarchy while holding __instanceLock__.
```
Comment (wsdwsd0829) [inline Source/ASDisplayNode.mm]: add space to trigger CI

**Pull request merged**